src/main/java/nl/novi/techiteasy1121/services/TelevisionService.java

```diff
@@ -1,72 +1,88 @@
-
-        for(Television tv : tvList) {
-            TelevisionDto dto = transferToDto(tv);
-            tvDtoList.add(dto);
-        }
-        return tvDtoList;
+        return transferTvListToDtoList(tvList);
     }
 
-    // Vanuit de repository kunnen we een lijst van Televisions met een bepaalde brand krijgen, maar de communicatie
-    // container tussen Service en Controller is de Dto. We moeten de Televisions dus vertalen naar TelevisionDtos. Dit
-    // moet een voor een, omdat de translateToDto() methode geen lijst accepteert als argument, dus gebruiken we een for-loop.
     public List<TelevisionDto> getAllTelevisionsByBrand(String brand) {
         List<Television> tvList = televisionRepository.findAllTelevisionsByBrandEqualsIgnoreCase(brand);
+        return transferTvListToDtoList(tvList);
+    }
+
+    public List<TelevisionDto> transferTvListToDtoList(List<Television> televisions){
         List<TelevisionDto> tvDtoList = new ArrayList<>();
 
-        for(Television tv : tvList) {
+        for(Television tv : televisions) {
             TelevisionDto dto = transferToDto(tv);
+            if(tv.getCiModule() != null){
+                dto.setCiModuleDto(ciModuleService.transferToDto(tv.getCiModule()));
+            }
+            if(tv.getRemoteController() != null){
+                dto.setRemoteControllerDto(remoteControllerService.transferToDto(tv.getRemoteController()));
+            }
             tvDtoList.add(dto);
         }
         return tvDtoList;
     }
 
-    // Deze methode is inhoudelijk hetzelfde als het was in de vorige opdracht. Wat verandert is, is dat we nu checken
-    // op optional.isPresent in plaats van optional.isEmpty en we returnen een TelevisionDto in plaats van een Television.
     public TelevisionDto getTelevisionById(Long id) {
-        Optional<Television> televisionOptional = televisionRepository.findById(id);
-        if (televisionOptional.isPresent()){
-            Television tv = televisionOptional.get();
+
+        if (televisionRepository.findById(id).isPresent()){
+            Television tv = televisionRepository.findById(id).get();
+            TelevisionDto dto =transferToDto(tv);
+            if(tv.getCiModule() != null){
+                dto.setCiModuleDto(ciModuleService.transferToDto(tv.getCiModule()));
+            }
+            if(tv.getRemoteController() != null){
+                dto.setRemoteControllerDto(remoteControllerService.transferToDto(tv.getRemoteController()));
+            }
+
             return transferToDto(tv);
         } else {
             throw new RecordNotFoundException("geen televisie gevonden");
         }
     }
 
-    // In deze methode moeten we twee keer een vertaal methode toepassen.
-    // De eerste keer van dto naar televsion, omdat de parameter een dto is.
-    // De tweede keer van television naar dto, omdat de return waarde een dto is.
     public TelevisionDto addTelevision(TelevisionInputDto dto) {
 
```

Comment:
Het gebruik van Optional bij het ophalen van de Television objecten is een goede praktijk om NullPointerExceptions te vermijden. Overweeg om expliciete foutafhandeling toe te voegen als een televisie niet gevonden wordt.